### PR TITLE
Update some out of date information

### PR DESCRIPTION
**Pull request by felicitymay**
This PR updates a few out of date references to Semmle and was prompted by reviewing https://github.com/Semmle/ql/pull/3226.

We don't distribute queries and libraries with the new generation of CodeQL tools, so I've also revised a couple of references to this.

### CODE REVIEW

File: docs/language/learn-ql/writing-queries/query-help.rst [hunk 1]
```diff
@@ -63,7 +63,7 @@ Section-level elements are used to group the information in the help file into s
 +--------------------+------------------------------------+------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
 | ``section``        | ``title`` Title of the section     | Any block element      | General-purpose section with a heading defined by the ``title`` attribute.                                                                    |
 +--------------------+------------------------------------+------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
-| ``semmleNotes``    | None                               | Any block element      | Semmle-specific notes about the query. This section is used only for queries that implement a rule defined by a third party. Default heading. |
+| ``semmleNotes``    | None                               | Any block element      | Implementation notes about the query. This section is used only for queries that implement a rule defined by a third party. Default heading. |
```
Comment (jf205) [inline docs/language/learn-ql/writing-queries/query-help.rst:66]:
```suggestion
| ``semmleNotes``    | None                               | Any block element      | Implementation notes about the query. This section is used only for queries that implement a rule defined by a third party. Default heading.  |
```

Comment (jf205) [reply docs/language/learn-ql/writing-queries/query-help.rst:66]: ☝️  Should fix the sphinx error.

**Comment by shati-patel** [comment]
👍🏼  apart from the table formatting issue (James beat me to it 😉)

**Comment by jf205** [comment]
Sorry, I forgot to 👀 the pinned message!

**Comment by felicitymay** [comment]
Thanks both of you

File: docs/language/learn-ql/writing-queries/query-help.rst [hunk 2]
```diff
@@ -84,7 +84,8 @@ The following elements are optional child elements of the ``section``, ``example
...
-   | ``include``    | ``src`` The query help file to include.                  | None               | Include a query help file at the location of this element. See :ref:`Query help inclusion <qhelp-inclusion>` below for more information.                                                                                                                                                                  |
+   | ``include``    | ``src`` The query help file to include.                  | None               | Include a query help file at the location of this element. See :ref:`Query help inclusion <
+   -inclusion>` below for more information.                                                                                                                                                                  |
```
Comment (shati-patel) [inline docs/language/learn-ql/writing-queries/query-help.rst:88]:
```suggestion
   | ``include``    | ``src`` The query help file to include.                  | None               | Include a query help file at the location of this element. See :ref:`Query help inclusion <qhelp-inclusion>` below for more information.                                                                                                                                                                  |
```

Sorry, I didn't spot the weird newline when reviewing!

Comment (felicitymay) [reply docs/language/learn-ql/writing-queries/query-help.rst]: Thanks - maybe I introduced it in the conflict resolution.

**Comment by felicitymay** [comment]
Many thanks for the reviews @shati-patel and @jf205. This probably should wait for the JavaScript and C/C++ teams to check they're happy with the changes to the library text.

**Comment by felicitymay** [comment]
I've merged the `rc/1.24` branch into my branch and retrageted the pull request. This has, unfortunately, removed all the approving review comment.

**Pull request merged**